### PR TITLE
Documentation Update[API-346]

### PR DESCRIPTION
**Pull request by yuce**
Documentation updates

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -129,3 +123,121 @@ func ExampleMap_Aggregate() {
...
+		panic(err)
+	}
+	if err := m.Delete(ctx, key); err != nil {
+		panic(err)
+	}
+	// you can use the subscriptionID later to remove the event listener.
+	if err := m.RemoveEntryListener(ctx, subscriptionID); err != nil {
+		panic(err)
+	}
+}
```
Comment (alisengul53) [inline example_test.go:171]: Could have more explanations like other examples.

File: proxy_map.go [hunk 2]
```diff
@@ -1097,49 +1152,60 @@ func (as attributeSet) Attrs() []string {
...
 func (c *MapEntryListenerConfig) NotifyEntryEvicted(enable bool) {
 	flagsSetOrClear(&c.flags, int32(EntryEvicted), enable)
 }
 
+// NotifyEntryExpired enables receiving an entry event when an entry is expired.
 func (c *MapEntryListenerConfig) NotifyEntryExpired(enable bool) {
 	flagsSetOrClear(&c.flags, int32(EntryExpired), enable)
 }
 
+// NotifyEntryAllEvicted enables receiving an entry event when an entry is evicted.
```
Comment (alisengul53) [inline proxy_map.go:1188]: Is it doing same thing with `NotifyEntryEvicted` method? Explenations are the same

**Review by alisengul53 — APPROVED** [review]

File: cluster/cluster_config.go [hunk 3]
```diff
@@ -25,19 +25,31 @@ import (
...
 const defaultName = "dev"
 
+// Config contains cluster and connection configuration.
 type Config struct {
-	loadBalancer       LoadBalancer
-	Security           SecurityConfig
-	Name               string `json:",omitempty"`
-	Cloud              CloudConfig
-	Network            NetworkConfig
+	loadBalancer LoadBalancer
```
[the remark this answers is not in the document]
Comment (yuce) [reply cluster/cluster_config.go:30]: Identifiers starting with lower case letters are unexported in Go,  so they are  private to the user.

File: proxy_map.go (hunk 2 above)
Comment (yuce) [reply proxy_map.go]: Thanks, fixed at https://github.com/hazelcast/hazelcast-go-client/pull/592/commits/db2c5ee518e06486f197a91fd252d442fc1de894

File: example_test.go (hunk 1 above)
Comment (yuce) [reply example_test.go]: Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/592/commits/284e2a54528a569c9dd4c14f3cf7cc66335cb051

**Review by alisengul53 — APPROVED** [review]

**Pull request merged**